ReactAndroid/src/main/java/com/facebook/react/uimanager/UIManagerModule.java

```diff
@@ -85,6 +91,21 @@ public UIManagerModule(
     super(reactContext);
     mEventDispatcher = new EventDispatcher(reactContext);
     DisplayMetrics displayMetrics = reactContext.getResources().getDisplayMetrics();
+
+    // Get the real display metrics if we are using API level 17 or higher.
+    // The real metrics include system decor elements (e.g. soft menu bar).
+    //
+    // See: http://developer.android.com/reference/android/view/Display.html#getRealMetrics(android.util.DisplayMetrics)
+    try {
+      if (Build.VERSION.SDK_INT >= 17) {
```

Comment:
@astreet 

Since `UIManagerModuleConstants` is using `DisplayMetricsHolder.getDisplayMetrics()` to set window width and height, we need to hold onto either the `Display` instance, or have a information holder object for the *real* width and height that accounts for API level 16 and 17+.

i.e. https://github.com/facebook/react-native/blob/7377fdc/ReactAndroid/src/main/java/com/facebook/react/uimanager/UIManagerModuleConstants.java#L115-L117

**Option #1**

We extend `DisplayMetricsHolder` with three methods:

1. `static void setRealDisplayDimension(int realWidth, int realHeight)`
2. `static int getRealDisplayWidth()`
3. `static int getRealDisplayHeight()`


**Option 2**

Create a new information holder class that holds the real width and height. We'll set these values from `UIManagerModule`, just like we are doing for `DisplayMetrics`.